src/mongoc/mongoc-cursor.c

```diff
@@ -229,6 +229,9 @@ _mongoc_cursor_new_with_opts (mongoc_client_t *client,
    cursor->client = client;
    cursor->is_command = is_command ? 1 : 0;
 
+   bson_init (&cursor->filter);
+   bson_init (&cursor->opts);
```

Comment:
This covers both code paths for basic cursor creation, since `_mongoc_cursor_new()` already always initialized these documents. This might also avoid possible crash in `_mongoc_cursor_cursorid_init()`, as that always attempts to destroy `cursor->filter` before copying the command to it. The only code path I spotted for that private function is through `mongoc_cursor_new_from_command_reply()`, which itself calls `_mongoc_cursor_new_with_opts()`, so I believe we're safe.

_mongoc_cursor_cursorid_init